### DISCUSSION TITLE
Download Excel data button

### DISCUSSION THREAD

**Discussion by salestial** [quarto-dev/quarto-cli#3752] [Q&A]
Festive greetings to all!

I have got a number of charts in my JupyterLab quarto document and would like to add a button for downloading the data behind the charts in Excel/XLSX (the data are substantial and need multiple worksheets). I have found ObservableJS code for adding a download button to save CSV formats (e.g. below) but nothing  for saving to Excel. Could someone assist please? Consider me a JavaScript novice.

Many thanks!

```
```{ojs}
// from https://observablehq.com/@jeremiak/download-data-button
button = (data, filename = 'data.csv') => {
  if (!data) throw new Error('Array of data required as first argument');

  let downloadData;
  if (filename.includes('.csv')) {
    downloadData = new Blob([d3.csvFormat(data)], { type: "text/csv" });
  } else {
    downloadData = new Blob([JSON.stringify(data, null, 2)], {
      type: "application/json"
    });
  }

  const size = (downloadData.size / 1024).toFixed(0);
  const button = DOM.download(
    downloadData,
    filename,
    `Download ${filename} (~${size} KB)`
  );
  return button;
}
```
```

**Comment by rgaiacs** [comment]
Welcome @salestial.

Because of the workflow used by Quarto to convert the Jupyter Notebook to HTML, I have built my CSV or Microsoft Excel files using Python, saved somewhere in the project (let's say `data_to_download`), and include a link to the files in my Jupyter Notebook.

> I have found ObservableJS code for adding a download button to save CSV formats (e.g. below) but nothing for saving to Excel.

Microsoft Excel is a more complex file than CSV and you might need to use a JavaScript library for that.
